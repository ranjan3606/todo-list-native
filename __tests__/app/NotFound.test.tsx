import React from 'react';
import { render } from '@testing-library/react-native';
import NotFoundScreen from '../../app/+not-found';

// Mock required modules
jest.mock('expo-router', () => ({
  Link: jest.fn(({ children }) => children),
  Stack: {
    Screen: jest.fn(() => null),
  },
}));

jest.mock('@expo/vector-icons', () => ({
  FontAwesome: 'FontAwesome',
}));

jest.mock('@/constants/Colors', () => ({
  Colors: {
    light: { 
      background: '#fff', 
      text: '#000', 
      tint: '#007AFF',
      lightText: '#666'
    },
    dark: { 
      background: '#000', 
      text: '#fff', 
      tint: '#0A84FF',
      lightText: '#999'
    },
    warning: '#f0ad4e',
  }
}));

jest.mock('@/services/theme', () => ({
  useActualColorScheme: () => 'light',
}));

jest.mock('@/i18n', () => ({
  useTranslation: () => ({
    t: (key) => {
      const translations = {
        'notFound.oops': 'Oops!',
        'notFound.title': 'Page Not Found',
        'notFound.description': 'The page you are looking for doesn\'t exist or has been moved.',
        'notFound.goBack': 'Go back home',
      };
      return translations[key] || key;
    }
  })
}));

describe('NotFoundScreen', () => {
  it('renders correctly', () => {
    const { toJSON, getByText } = render(<NotFoundScreen />);
    
    expect(toJSON()).toBeTruthy();
    expect(getByText('Page Not Found')).toBeTruthy();
    expect(getByText('The page you are looking for doesn\'t exist or has been moved.')).toBeTruthy();
    expect(getByText('Go back home')).toBeTruthy();
  });
  
  it('matches snapshot', () => {
    const { toJSON } = render(<NotFoundScreen />);
    expect(toJSON()).toMatchSnapshot();
  });
});