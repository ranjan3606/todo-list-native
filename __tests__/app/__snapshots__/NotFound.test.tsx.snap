// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundScreen matches snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 20,
      },
      {
        "backgroundColor": "#fff",
      },
    ]
  }
>
  <View
    style={
      {
        "marginBottom": 20,
      }
    }
  >
    <FontAwesome
      color="#f0ad4e"
      name="exclamation-triangle"
      size={64}
    />
  </View>
  <Text
    style={
      [
        {
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 10,
        },
        {
          "color": "#000",
        },
      ]
    }
  >
    Page Not Found
  </Text>
  <Text
    style={
      [
        {
          "marginBottom": 20,
          "textAlign": "center",
        },
        {
          "color": "#666",
        },
      ]
    }
  >
    The page you are looking for doesn't exist or has been moved.
  </Text>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "fontWeight": "500",
        },
        {
          "color": "#007AFF",
        },
      ]
    }
  >
    Go back home
  </Text>
</View>
`;
